### PR TITLE
Fix wrong render script string cookie

### LINKED ISSUE #31385: Secure Html renderer wrong window.cookiesConfig
### Preconditions (*)

Magento 2.4.1

### Steps to reproduce (*)

vendor/magento/module-cookie/view/base/templates/html/cookie.phtml

### Expected result (*)

```
<script>
    window.cookiesConfig = window.cookiesConfig || {};
    window.cookiesConfig.secure = true;
</script>
```

### Actual result (*)

```
<script>true</script>
```

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

### PR DESCRIPTION
**Pull request by mrtuvn**
### Description (*)

Correct mistake in core when developer previously not properly use string script

Before patch
Screen Shot 2020-12-22 at 15 44 18

After this patch 

Screen Shot 2020-12-22 at 15 39 36

This issue can reproduce in 2.4.0, 2.4.1 official and 2.4-develop branch

### Fixed Issues (if relevant)

1. Fixes magento/magento2#31385

### Manual testing scenarios (*)

1. View page source
2. Search config script window.cookiesConfig
3. Before this patch magento render incorrect (only show content script true). So we can't search this

### Questions or comments

CC: @peterjaap i found you have issue reported relate with cookies
This PR can affect to cookie area too
@bubasuma 
Turn out when digging this problem i see still some remain problem among core codebase

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by ihor-sviziev** [comment]
@magento run Functional Tests B2B, Functional Tests CE, Functional Tests EE

**Comment by ihor-sviziev** [comment]
Seems like MHI failure is false positive

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Could you cover your changes with any kind of automated test?

**Comment by mrtuvn** [comment]
@magento give me 2.4-develop instance

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run Semantic Version Checker, Magento Health Index, Integration Tests, Functional Tests CE, Functional Tests EE, Functional Tests B2B

**Comment by bubasuma** [comment]
Hi @mrtuvn,
This issue was fixed in 2.4.2 and is not reproducible in 2.4-develop.
Please refer to commit https://github.com/magento/magento2/commit/d04e4d505aced753aedadf1f3593ef5420825857

**Comment by mrtuvn** [comment]
Hmm seem it's not work when i check in latest code branch

File: app/code/Magento/Cookie/Test/Mftf/Test/StorefrontVerifyUnsecureCookieTest.xml [hunk 1]
```diff
@@ -21,16 +21,28 @@
             <group value="configuration"/>
         </annotations>
         <before>
+            <!-- Go to storefront home page -->
+            <actionGroup ref="StorefrontOpenHomePageActionGroup" stepKey="openStoreFrontHomePage"/>
+            <executeJS function="return window.location.host" stepKey="hostname"/>
+            <!-- Login as admin -->
+            <actionGroup ref="AdminLoginActionGroup" stepKey="loginAsAdmin"/>
+            <magentoCLI command="config:set web/unsecure/base_url https://{$hostname}/" stepKey="setUnsecureBaseURL"/>
```
Comment (bubasuma) [inline app/code/Magento/Cookie/Test/Mftf/Test/StorefrontVerifyUnsecureCookieTest.xml:29]: Please pay attention to the description of the test case. it says "Verify that cookie are not secure on storefront over **http**". Please add another test if you want to cover different scenario. As I see you probably want to test the scenario when https is configured but use_in_frontend is set 0.

File: app/code/Magento/Cookie/Test/Mftf/Test/StorefrontVerifySecureCookieTest.xml [hunk 2]
```diff
@@ -22,26 +22,24 @@
...
             <magentoCLI command="config:set web/secure/use_in_frontend 1" stepKey="useSecureURLsOnStorefront"/>
             <actionGroup ref="CliCacheFlushActionGroup" stepKey="flushCache">
-                <argument name="tags" value=""/>
+                <argument name="tags" value="config full_page"/>
             </actionGroup>
         </before>
         <after>
-            <amOnPage url="/" stepKey="goToHomePage"/>
-            <executeJS function="return window.location.host" stepKey="hostname"/>
-            <magentoCLI command="config:set web/unsecure/base_url http://{$hostname}/" stepKey="setUnsecureBaseURL"/>
```
Comment (bubasuma) [inline app/code/Magento/Cookie/Test/Mftf/Test/StorefrontVerifySecureCookieTest.xml:37]: could you please explain why you removed these steps? aren't they suppose to revert the settings set in the "before" tag?

**Review by bubasuma — COMMENTED** [review]
@mrtuvn 
It's strange. I cannot reproduce it in 2.4-develop.

**Comment by mrtuvn** [comment]
I will revert change in mftf and find another way for this scenario

**Comment by ihor-sviziev** [comment]
@mrtuvn seems like @bubasuma mean that this issue was already fixed in 2.4-develop. Could you check that?

**Comment by mrtuvn** [comment]
When try again one more i can't  reproduce problem in 2.4-develop branch so i will close this!

**Pull request closed without merging**